src/sparsezoo/search.py

```diff
@@ -0,0 +1,130 @@
+        the object belongs to, e.g. base, pruned, pruned_quant
+    :param sparse_category: The degree of sparsification of the model the object
+        belongs to; e.g. none, conservative (~100% baseline metric),
+        moderate (>=99% baseline metric), aggressive (<99% baseline metric)
+    :param sparse_target: The deployment target of sparsification of the model
+        the object belongs to; e.g. edge, deepsparse, deepsparse_throughput, gpu
+    :param release_version: The sparsezoo release version for the model
+    :param page: the page of values to get
+    :param page_length: the page length of values to get
+    :param override_folder_name: Override for the name of the folder to save
+        this file under
+    :param override_parent_path: Path to override the default save path
+        for where to save the parent folder for this file under
+    :param force_token_refresh: True to refresh the auth token, False otherwise
+    :return: The requested list of Model instances
+    """
+    args = {
+        "domain": domain,
+        "sub_domain": sub_domain,
+        "architecture": architecture,
+        "sub_architecture": sub_architecture,
+        "framework": framework,
+        "repo": repo,
+        "dataset": dataset,
+        "training_scheme": training_scheme,
+        "sparse_name": sparse_name,
+        "sparse_category": sparse_category,
+        "sparse_target": sparse_target,
+        "release_version": release_version,
+    }
+
+    args = {k: v for k, v in args.items() if v is not None}
+
+    logging.debug(f"Search_models: searching models with args {args}")
+    response_json = search_model_get_request(
+        args=args,
+        page=page,
+        page_length=page_length,
+        force_token_refresh=force_token_refresh,
+    )
+
+    return [
+        Model(model_dict_to_stub(model_dict)) for model_dict in response_json["models"]
+    ]
+
+
+def model_dict_to_stub(model_dict: Dict[str, Optional[str]]) -> str:
+
+    domain = model_dict.get("domain")
+    sub_domain = model_dict.get("sub_domain")
+    architecture = model_dict.get("architecture")
+    sub_architecture = model_dict.get("sub_architecture")
+    framework = model_dict.get("framework")
+    repo = model_dict.get("repo")
+    dataset = model_dict.get("dataset")
+    sparse_tag = model_dict.get("sparse_tag")
+
+    if sub_architecture is not None:
+        sub_architecture = '-' + sub_architecture
+
+    stub = f"zoo:{domain if domain is not None else ''}/" \
+           f"{sub_domain if sub_domain is not None else ''}/" \
+           f"{architecture if architecture is not None else ''}" \
+           f"{sub_architecture if sub_architecture is not None else ''}/" \
+           f"{framework if framework is not None else ''}/" \
+           f"{repo if repo is not None else ''}/" \
+           f"{dataset if dataset is not None else ''}/" \
+           f"{sparse_tag if sparse_tag is not None else ''}"
```

Comment:
So if a bunch of these are none would it just have a lot of slashes next to each other?

`zoo://////`?

Reply:
Technically yes.